packages/flutter_tools/lib/src/ios/plist_utils.dart

```diff
@@ -2,39 +2,46 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import '../base/context.dart';
 import '../base/file_system.dart';
 import '../base/process.dart';
+import '../convert.dart';
+import '../globals.dart';
 
-const String kCFBundleIdentifierKey = 'CFBundleIdentifier';
-const String kCFBundleShortVersionStringKey = 'CFBundleShortVersionString';
-const String kCFBundleExecutable = 'CFBundleExecutable';
-
-// Prefer using [iosWorkflow.getPlistValueFromFile] to enable mocking.
-String getValueFromFile(String plistFilePath, String key) {
-  // TODO(chinmaygarde): For now, we only need to read from plist files on a mac
-  // host. If this changes, we will need our own Dart plist reader.
-
-  // Don't use PlistBuddy since that is not guaranteed to be installed.
-  // 'defaults' requires the path to be absolute and without the 'plist'
-  // extension.
-  const String executable = '/usr/bin/defaults';
-  if (!fs.isFileSync(executable))
-    return null;
-  if (!fs.isFileSync(plistFilePath))
-    return null;
-
-  final String normalizedPlistPath = fs.path.withoutExtension(fs.path.absolute(plistFilePath));
-
-  try {
-    final List<String> args = <String>[
-      executable, 'read', normalizedPlistPath,
-    ];
-    if (key != null && key.isNotEmpty) {
-      args.add(key);
+class PlistUtils {
+  const PlistUtils();
+
+  static const String kCFBundleIdentifierKey = 'CFBundleIdentifier';
+  static const String kCFBundleShortVersionStringKey = 'CFBundleShortVersionString';
+  static const String kCFBundleExecutable = 'CFBundleExecutable';
+
+  static PlistUtils get instance => context.get<PlistUtils>() ?? const PlistUtils();
+
+  Map<String, dynamic> parseFile(String plistFilePath) {
+    assert(plistFilePath != null);
+    const String executable = '/usr/bin/plutil';
+    if (!fs.isFileSync(executable))
+      return null;
```

Comment:
Should this throw a tool exit instead, so we can find out that it's happening?

Reply:
Yeah.  Done.

Reply:
Ah, I see now - we had at least one code path that was calling this from Linux (in project.dart).

I updated it to do the following:

1. Changed it to throw a newly defined `FileNotFoundException`, specifically distinct from `ToolExit` so that it will show up in our analytics as a tool crash and alert us to something wrong with our code.
2. Wrap the call site in `project.dart` in a try/catch.  There's a separate discussion about whether it should be refactored so as to not try to do anything plist related on Linux, but it at least looks like it was intentionally set up that way, so I'm not going to tug on that thread in this PR.